circuit-benchmarks/build.rs

```diff
@@ -0,0 +1,130 @@
+        const MEMORY_ROWS_MAX: usize = {};
+        const MEMORY_ADDRESS_MAX: usize = {};
+        const STACK_ROWS_MAX: usize = {};
+        const STACK_ADDRESS_MAX: usize = {};
+        const STORAGE_ROWS_MAX: usize = {};
+        const GLOBAL_COUNTER_MAX: usize = {};
+
+        let public_inputs_size = 0;
+        let empty_circuit = StateCircuit::<
+            GLOBAL_COUNTER_MAX,
+            MEMORY_ROWS_MAX,
+            MEMORY_ADDRESS_MAX,
+            STACK_ROWS_MAX,
+            STACK_ADDRESS_MAX,
+            STORAGE_ROWS_MAX,
+        >::default();
+
+        // Initialize the polynomial commitment parameters
+        let rng = XorShiftRng::from_seed([
+            0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32,
+            0x54, 0x06, 0xbc, 0xe5,
+        ]);
+
+        // Bench setup generation
+        let setup_message = format!(\"Setup generation with degree = {{}}\", DEGREE);
+        let start1 = start_timer!(|| setup_message);
+        let params = Setup::<Bn256>::new(DEGREE as u32, rng);
+        let verifier_params =
+            Setup::<Bn256>::verifier_params(&params, public_inputs_size).unwrap();
+        end_timer!(start1);
+
+        // Initialize the proving key
+        let vk =
+            keygen_vk(&params, &empty_circuit).expect(\"keygen_vk should not fail\");
+        let pk = keygen_pk(&params, vk, &empty_circuit)
+            .expect(\"keygen_pk should not fail\");
+        // Create a proof
+        let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
+
+        // Bench proof generation time
+        let proof_message = format!(\"State Proof generation with {{}} rows\", DEGREE);
+        let start2 = start_timer!(|| proof_message);
+        create_proof(&params, &pk, &[empty_circuit], &[&[]], &mut transcript)
+            .expect(\"proof generation should not fail\");
+        let proof = transcript.finalize();
+        end_timer!(start2);
+
+        // Bench verification time
+        let start3 = start_timer!(|| \"State Proof verification\");
+        let mut transcript = Blake2bRead::<_, _, Challenge255<_>>::init(&proof[..]);
+
+        verify_proof(&verifier_params, pk.get_vk(), &[&[]], &mut transcript)
+            .expect(\"failed to verify bench circuit\");
+        end_timer!(start3);
+    }}
+}}",
+        deps,
+        degree,
+        memory_rows_max,
+        memory_address_max,
+        stack_rows_max,
+        stack_address_max,
+        storage_rows_max,
+        global_counter_max
+    );
+
+    let mut state_file = File::create("src/state_circuit.rs")
+        .expect("Error generating state_circ.rs file");
+    state_file
+        .write(&_bench_code.as_bytes())
+        .expect("Error writing to state_circ.rs file");
+
+    // Add state_circuit module to `lib.rs`
+    let state_mod = r#"
+#[cfg(feature = "benches")]
+pub mod state_circuit;
+"#;
+    let mut lib_file = OpenOptions::new()
+        .write(true)
+        .append(true)
```

Comment:
Won't this append the same lines every build, extending the `lib.rs` file indefinitely?  I think `lib.rs` should be static and not touched during `build.rs`

Reply:
Yeah I've just tried running the `DEGREE=4 make evm_bench` multiple times and on each one, the following is appended to `lib.rs`:
```rust

#[cfg(feature = "benches")]
pub mod state_circuit;

```

Reply:
Hmmm. I can add a check for that.

Why I did this was basically because otherways there are some checks like `rustmft` that do not compile the libs. And those fail since they see a module definition but when jump into it, they fail.

Reply:
Ok, now I understand why you added the `build` step in the rustfmt job in the github action!
Wouldn't making a build before rustfmt solve the problem of the files not existing?

Reply:
After looking around, it seems there's a lighter way to trigger the `build.rs`: instead of `cargo build`, running `cargo check` will also do the trick.